### DISCUSSION TITLE
Reverse proxy via caddy a piu servizi

### DISCUSSION THREAD

**Discussion by Jamiro75** [indomus/forum#175] [GENERALE - Discussioni generiche]
Una quesito:
con i vari esempi in rete non sono riuscito ad eseguire questa cosa :
configurando un solo servizio nella fattispecie HA  che sara raggiungibile quindi sulla classica 443  tutto funziona come nel vostro esempio:

```
https://voxxxxx.vodafone.it { 
  import https_header
  reverse_proxy http://IP_DELLA_ISTANZA_HOME_ASSISTANT:8123
}
```

e qui tutto funziona anche benone  usando il mio indirizzo FQDN della vodafone station al posto di usare duckdns o simili.
 non sono riuscito ad inserire un altro servizio usando la stessa porta esterna es la 443 senza aprire altre porte, la mia idea era se si poteva creare  dei percorsi tipo :

https://voxxxxx.vodafone.it/ha 
 reverse_proxy http://IP_DELLA_ISTANZA_HOME_ASSISTANT:8123

https://voxxxxx.vodafone.it/nodered 
 reverse_proxy http://IP_DELLA_ISTANZA_NODERED:1880

altri......
ecc...

ho trovato esempi ... ma non sono stato in grado .... 
c'e' qualcuno che ha fatto qualcosa del genere ?

grazie mille in anticipo 
Simone.

**Comment by snakuzzo** [comment]
Definire servizi su base `path` non è detto che funzioni sempre. Dipende anche da com'è scritta l'applicazione...potrebbero esserci casini.
In linea di massima dovresti poterlo fare come hai indicato anche tu...

```
https://voxxxxx.vodafone.it/homeassistant { 
  import https_header
  reverse_proxy http://192.168.1.10:8123
}

https://voxxxxx.vodafone.it/nodered { 
  reverse_proxy http://192.168.1.10:1880
}
```

Se non funziona devi guardare i log di Caddy (quelli relativi ai singoli servizi ... che dovrai abilitarli) e/o i log delle singole applicazioni esposte
